src/action.js

```diff
@@ -49,3 +76,41 @@ export async function run() {
       `)
   }
 }
+
+async function uploadMedia(client, media, mediaAltText) {
+  if (media?.length) {
+    core.info(`Twitter upload media: ${media.join('; ')}`)
+
+    const mediaIds = await Promise.all(
+      media.map(media => client.v1.uploadMedia(media))
+    )
+    core.info(`Twitter upload completed - mediaId: ${mediaIds.join('; ')}`)
+
+    if (mediaAltText?.length) {
+      try {
+        await Promise.all(
+          mediaIds.map((mediaId, index) => {
+            core.info(
+              `Twitter createMediaMetadata - mediaId: 
+            ${mediaId} ; alt-text: ${mediaAltText[index]}`
+            )
+            if (!mediaAltText?.[index]?.trim()) return Promise.resolve()
+            return client.v1.createMediaMetadata(mediaId, {
+              alt_text: { text: mediaAltText[index] }
+            })
+          })
+        )
+      } catch (err) {
+        core.warning(
+          `Twitter createMediaMetadata - Failed. ${err} ${
+            err.data ?? ''
+          }`.trim()
+        )
+      }
+    }
+    return mediaIds
+  } else {
+    core.info(`Twitter no media to upload`)
+    return null
```

Comment:
no need to return null. also, prefer returning early, meaning, invert the condition of these 2 if/else branches